### ISSUE TITLE
Clippy error "useless lint attribute" when using FromPrimitive

### ISSUE THREAD

**Issue by dignifiedquire** [rust-num/num-derive#20]
Getting this error when running latest clippy on my code, after adding `FromPrimitive`

```
error: useless lint attribute
  --> src/packet/tags/sig.rs:15:1
   |
15 | #[derive(FromPrimitive)]
   | ^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: for further information visit https://rust-lang-nursery.github.io/rust-clippy/v0.0.212/index.html#useless_attribute
```

code is just a simple enum like this

```rust
#[derive(FromPrimitive)]
pub enum KeyFlag {
   First = 1,
   Second = 2,
}
```

- rust version: `rustc 1.32.0-nightly (25a42b2ce 2018-11-07)`
- clippy version: `clippy 0.0.212 (d8b42690 2018-11-04)`
- num-derive: `0.2.3`

**Comment by cuviper** [comment]
> ### What it does
> Checks for extern crate and use items annotated with lint attributes

I think it's warning about this code in the expansion:

```rust
    #[allow(unused_qualifications)]
    extern crate num_traits as _num_traits;
```

That `allow` was added for #9, but at the time it was placed on the "dummy" `const`.  It was moved onto this `extern crate` in #17.  I need to make sure of what it was meant to protect exactly -- as-is, the `issue-9` test **fails** on Rust 1.16 through 1.28, but 1.15 and 1.29+ are fine with it.

**Comment by athei** [comment]
What is necessary to fix this? The "useless lint attribute" is in the correctness category of clips and defaults to deny.

**Comment by cuviper** [comment]
@athei probably just needs to move the lint back onto the `dummy_const`.

**Comment by cuviper** [comment]
Should be fixed by num-derive 0.2.5. Let me know if there's still an issue.

**Comment by athei** [comment]
Indeed that fixed the warning.
